### ISSUE TITLE
The data source api returned a lot of data

### ISSUE THREAD

**Issue by yylbfyl** [grafana/grafana#74713]
### What happened?

We have two different version Grafana , 7.5.11 and 9.3.6 , we found a panel in dashboard is load too slow on version 9.3.6 but it very fast on 7.5.11. 
The Prometheus datasource and dashboard variables are the same , In Query Inspect the number of queries are the same ,  through comparison I find the query API is different , the version 7.5.11 used API  "/api/datasources/proxy/1/api/v1/query_range"  to query and return 26KB data , but the version 9.3.6 used API "/api/ds/query" to query and return 400MB data , so the loading of the panel is very slow. 

I try to import the same dashboard and datasource into Grafana latest  version 10.1.1 , the problem remains the same, so I think the reason is the different API between the different Grafana version , I down't know why the API "/api/ds/query" returned a lot of data .
[![pP2i5Ke.md.png](https://s1.ax1x.com/2023/09/12/pP2i5Ke.md.png)](https://imgse.com/i/pP2i5Ke)
[![pP2iIDH.md.png](https://s1.ax1x.com/2023/09/12/pP2iIDH.md.png)](https://imgse.com/i/pP2iIDH)

### What did you expect to happen?

I think the API "/api/ds/query" should retuen the same data like "/api/datasources/proxy/1/api/v1/query_range" , a lot of data lead to the panel loading so slow.
If there is any way to use old API "/api/datasources/proxy/1/api/v1/query_range" through the configuration in Grafana , I'd like to change it.

### Did this work before?

Version 7.5.11 works fine, version 9.3.6 and 10.1.1 have the problem.

### How do we reproduce it?

1.Find a Prometheus datasource that contain kafka mertics kafka_cluster_partition_replicascount
2.Created a panel in dashboard, query "sum(kafka_cluster_partition_replicascount{topic =~ "$topic"}) by (topic)/3" 

### Environment (with versions)?

Grafana: 9.3.6
OS: Windows 10
Browser: Chrome 116.0.5845.111 (Official Build) (64-bit)

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

**Comment by mellieA** [comment]
@yylbfyl I've tested this recently with Grafana 10.2 and it uses the query endpoint there. 

What happens if you toggle to use the "instant" query option?

**Comment by yylbfyl** [comment]
Hello @mellieA , thanks for your reply , I toggle to use the "instant" query option , the total number rows decreased , but the query api still returned 400MB data.
[![pPRw1wF.md.png](https://z1.ax1x.com/2023/09/14/pPRw1wF.md.png)](https://imgse.com/i/pPRw1wF)

**Comment by mellieA** [comment]
@yylbfyl Thank you for trying it. Are both Grafana's connected to the same prometheus datasource?

**Comment by itsmylife** [comment]
We use the same endpoint in the backend as well. See here: https://github.com/grafana/grafana/blob/main/pkg/tsdb/prometheus/client/client.go#L45

Could you please share a `.har` file that you get from Grafana`v7.5.x` and Grafana `v10.1.x`. 
It would be super helpful if you switch between `instant` and `time series` while recording your `.har` file. 
Also could you please share your prometheus datasource configuration? 

Additionally you can do it via `Get Help` menu on a dashboard panel. Follow this link for more information https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/

![Image](https://github.com/grafana/grafana/assets/820946/b9a72230-d8e8-452c-93c2-5c933c53e7e2)

**Comment by yylbfyl** [comment]
> We use the same endpoint in the backend as well. See here: https://github.com/grafana/grafana/blob/main/pkg/tsdb/prometheus/client/client.go#L45
> 
> Could you please share a `.har` file that you get from Grafana`v7.5.x` and Grafana `v10.1.x`. It would be super helpful if you switch between `instant` and `time series` while recording your `.har` file. Also could you please share your prometheus datasource configuration?
> 
> Additionally you can do it via `Get Help` menu on a dashboard panel. Follow this link for more information https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/
> 
> ![Image](https://user-images.githubusercontent.com/820946/268935099-b9a72230-d8e8-452c-93c2-5c933c53e7e2.png)

Hello @itsmylife , thanks your reply , I collected the Grafana 10.1.1 debug file and .har file of both 10.1.1 and 7.5.11 , added prometheus.yaml file in zip , please have a look , thanks. http://test.iooi.in/debug-grafana.zip

**Comment by itsmylife** [comment]
@yylbfyl I checked the files. Thanks a lot for them. Here are my findings:

1. I think you are not leveraging the `gzip` option. You might want to enable it. See: https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#enable_gzip
2. Even if you enable `gzip` on the frontend you will receive the same amount of data which means there will be a data processing price. 
3. I see the same amount of rows are returning for `v7.x` and `v9.x`. Since the executed query is so big and we are adding that information to each dataframe the response size is getting huge. 
4. That was a known issue and we changed the behaviour recently. See: https://github.com/grafana/grafana/pull/73678 this will be available with `v10.2.0` cc: @kylebrandt 

My Suggestions:
- Upgrade to `10.2.0` when It will be available (~1 month)
- Enable gzip option. More Info: https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#enable_gzip

**Comment by yylbfyl** [comment]
Hello @itsmylife , thanks for your analysis , when I enable gzip option , the transferred data decreased to 58MB , it's a good news , but total resource still over 400MB , it still more than version 7.5.11 that total data only 16MB , is it a known issue ? #73678  

[![pPILGgU.png](https://z1.ax1x.com/2023/09/21/pPILGgU.png)](https://imgse.com/i/pPILGgU)
[![pPILUb9.jpg](https://z1.ax1x.com/2023/09/21/pPILUb9.jpg)](https://imgse.com/i/pPILUb9)

**Comment by itsmylife** [comment]
Yes it was a known thing. We add the executed query string to every frame previously. Generally executed query string is short and that doesn't hurt too much (which doesn't mean it is correct to add it to every frame). But in your case, the query string is 128653 characters long. But we changed that. See https://github.com/grafana/grafana/pull/73678
This will be available with v10.2.0

**Comment by yylbfyl** [comment]
Thanks a lot ,I will waiting for v10.2.0 release to solve this problem.
